controller/utils.go

```diff
@@ -0,0 +1,46 @@
+package controller
+
+import (
+	longhorn "github.com/longhorn/longhorn-manager/k8s/pkg/apis/longhorn/v1beta2"
+	"github.com/longhorn/longhorn-manager/types"
+)
+
+func hasReplicaEvictionRequested(rs map[string]*longhorn.Replica) bool {
+	for _, r := range rs {
+		if r.Status.EvictionRequested {
+			return true
+		}
+	}
+
+	return false
+}
+
+func isVolumeMigrating(v *longhorn.Volume) bool {
+	return v.Spec.MigrationNodeID != "" || v.Status.CurrentMigrationNodeID != ""
+}
+
+func (vc *VolumeController) isVolumeUpgrading(v *longhorn.Volume) bool {
+	return v.Status.CurrentImage != v.Spec.EngineImage
+}
+
```

Comment:
Please move this method to the right file, volume_controller.go for better code placement. Also, other methods if needed.

Reply:
@shuo-wu recommend bring non-method functions that are used by multiple controller out to `util.go` file. I think it would be nicer that way. WDYT?